// Copyright 2024 The JAX SC Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package jax_sc_embedding;

message AllReduceData {
  int32 sync_key = 1;
  int32 src_rank = 2;
  oneof value {
    bool bool_val = 3;
    uint64 uint64_val = 4;
  }
}

message AllReduceResponse {}

service AllReduceGrpcService {
  // Unary RPC for a host to send its data to another host as part of an
  // All-to-All exchange.
  rpc ContributeData(AllReduceData) returns (AllReduceResponse) {}
}
